infra/token-transfer-server/src/tasks/transfer.js

```diff
@@ -27,43 +40,39 @@ const executeTransfers = async () => {
         { transaction: txn }
       )
       if (outstandingTasks.length > 0) {
-        throw new Error(
-          `Found incomplete transfer task(s), wait for completion or clean up manually.`
-        )
+        logger.warn(`Found incomplete transfer task(s), unable to proceed.`)
+        return false
       }
 
-      const waitingTransfers = await Transfer.findAll(
+      const processingTransfers = await Transfer.findAll(
         {
           where: {
-            [Sequelize.Op.or]: [
-              {
-                status: enums.TransferStatuses.WaitingConfirmation
-              },
-              {
-                status: enums.TransferStatuses.Processing
-              }
-            ]
+            status: enums.TransferStatuses.Processing
           }
         },
         { transaction: txn }
       )
-      if (waitingTransfers.length > 0) {
-        throw new Error(
-          `Found unconfirmed transfer(s). Fix before running this script again.`
-        )
+      if (processingTransfers.length > 0) {
+        logger.warn(`Found processing transfers, unable to proceed`)
+        return false
       }
 
-      transferTask = await TransferTask.create(
+      const now = moment.utc()
+      return await TransferTask.create(
         {
-          start: moment.utc()
+          start: now,
+          created_at: now,
+          updated_at: now
         },
         { transaction: txn }
       )
     }
   )
 
+  if (!transferTask) return
+
   const cutoffTime = moment.utc().subtract(largeTransferDelayMinutes, 'minutes')
-  const transfers = await Transfer.findAll({
+  const transfer = await Transfer.findOne({
```

Comment:
should we add an order by created_at asc so that we process in fifo order?

Reply:
👍